docs/howto/manage-libraries.md

```diff
@@ -127,42 +125,47 @@ def my_charm_type(endpoint):
 
 @pytest.fixture
 def context(my_charm_type):
-    return Context(my_charm_type, meta=my_charm_type.META)
+    return testing.Context(my_charm_type, meta=my_charm_type.META)
 
 
 def test_charm_runs(context):
     """Verify that the charm executes regardless of how we name the requirer endpoint."""
-    # arrange
-    state_in = State()
-    # act
-    context.run('start', state_in)
+    # Arrange:
+    state_in = testing.State()
+    # Act:
+    context.run(context.on.start(), state_in)
 
 
 @pytest.mark.parametrize('n_relations', (1, 2, 7))
 def test_charm_runs_with_relations(context, endpoint, n_relations):
     """Verify that the charm executes when there are one or more relations on the endpoint."""
-    # arrange
-    state_in = State(relations=[
-        Relation(endpoint=endpoint, interface='my-interface', remote_app_name=f"remote_{n}") for n in range(n_relations)
-    ])
-    # act
-    state_out = context.run('start', state_in)
-    # assert
+    # Arrange:
+    state_in = testing.State(relations={
+        testing.Relation(
+            endpoint=endpoint,
+            interface='my-interface',
+            remote_app_name=f"remote_{n}",
+        )
+        for n in range(n_relations
+    })
+    # Act:
+    state_out = context.run(context.on.start(), state_in)
+    # Assert:
     for relation in state_out.relations:
         assert not relation.local_app_data  # remote side didn't publish any data.
 
 
 @pytest.mark.parametrize('n_relations', (1, 2, 7))
 def test_relation_changed_behaviour(context, endpoint, n_relations):
     """Verify that the charm lib does what it should on relation changed."""
-    # arrange
-    relations = [Relation(
+    # Arrange:
+    relations = {Relation(
         endpoint=endpoint, interface='my-interface', remote_app_name=f"remote_{n}",
         remote_app_data={"foo": f"my-data-{n}"}
-    ) for n in range(n_relations)]
-    state_in = State(relations=relations)
+    ) for n in range(n_relations)}
```

Comment:
nit: is our ruff happy with this style? Worth putting the outer braces on their own lines?

Reply:
> nit: is our ruff happy with this style? Worth putting the outer braces on their own lines?

I'm not sure what formatting has been done here. With the docs now in git, I'd like to have some auto-formatting and testing (like the Scenario README has) so that we can keep these clean, but that's something for another (rainy) day.

I wasn't really intending to improve these docs, just migrate them away from Harness and Scenario < 7, but it would be easy enough to do this, sure.

Reply:
No need to make changes here if auto-formatting is already on your radar for a rainy day